src/dxvk/dxvk_memory.h

```diff
@@ -17,8 +17,26 @@ namespace dxvk {
     VkDeviceSize memoryAllocated = 0;
     VkDeviceSize memoryUsed      = 0;
   };
-  
-  
+
+
+  /**
+   * \brief Shared handle info
+   *
+   * The shared resource information for a given resource.
+   * IMPORT:     type = [handle type], handle = [shared handle]
+   * EXPORT:     type = [handle type], handle = nullptr
+   * NOT SHARED: type = [VK_EXTERNAL_MEMORY_HANDLE_TYPE_FLAG_BITS_MAX_ENUM], handle = anything
+   */
+  struct DxvkSharedHandleInfo {
+    VkExternalMemoryHandleTypeFlagBits type   = VK_EXTERNAL_MEMORY_HANDLE_TYPE_FLAG_BITS_MAX_ENUM;
+    HANDLE                             handle = INVALID_HANDLE_VALUE;
```

Comment:
Hardcoding win32 types here is going to cause a lot of pain and suffering for dxvk-native work, this should really be a union type with `HANDLE` being `#ifdef`'d out based on whether we're compiling for windows. Perhaps add a `void* dummy;` field or something to make it compile on non-win32.

All shared handle interactions in the backend should be built around that so we can easily ifdef out the win32 parts and perhaps support something like host memory import, should it become necessary.

It's not necessary to make the whole thing compile on native straight away, but we shouldn't make it harder than necessary.

Reply:
Let me know if the changes I made are enough on this front.